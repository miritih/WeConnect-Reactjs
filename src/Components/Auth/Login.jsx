import React, { Component } from 'react';
// import components 
import LoginForm from '../forms/Login';
import NavBar from '../NavBar';

class Login extends Component {


	render() {
		return (
			<div>
				<NavBar history={this.props.history}
					location={this.props.location}
				/>
				<LoginForm history={this.props.history} />
			</div>
		);
	}
}

export default Login;
