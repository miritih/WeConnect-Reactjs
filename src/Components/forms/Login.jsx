import React, { Component } from 'react';
import Authservice from '../Auth/AuthService';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import InputField from '../inputs/InputField';

class LoginForm extends Component {
	constructor() {
		super();
		this.state = {
			username: '',
			password: ''
		};
		this.auth = new Authservice();
		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(e) {
		this.setState({
			[e.target.name]: e.target.value
		});
	}
	handleSubmit(e) {
		e.preventDefault();
		this.auth.login(this.state.username, this.state.password)
			.then(res => {
				if (res.status >= 200 && res.status < 300) {
					this.props.history.replace('/');
					toast.success(({ closeToast }) => <div>
						<h3>Success</h3>
						<p>Logged in successfully</p>
					</div>);
				}
				else {
					this.setState({
						username: '',
						password: ''
					});
					// create the error message container
					// this willl be shown in our notification
					toast.error(({ closeToast }) =>
						<div>
							<h3>Opps!!</h3>
							<p>Wrong Username or password entered</p>
						</div>
					);
				}
			})
			.catch(err => {
				toast.error(({ closeToast }) => <div>
					<h3>Opps!!</h3>
					<p>Sorry! Something went wrong. If the problem persist, contact support</p>
				</div>);
			});
	}
	componentWillMount() {
		if (this.auth.isLoggedIn())
			this.props.history.replace('/');
	}
	render() {
		return (
			<div className="container wrapper">
				<div className="main-center">
					<div className="card">
						<div className="card-body">
							<img className="profile-img" src="//ssl.gstatic.com/accounts/ui/avatar_2x.png" alt="login" />
							<p id="profile-name" className="profile-name-card"></p>
							<form className="form" onSubmit={this.handleSubmit}>
								<InputField
									type='text'
									name='username'
									label='Username'
									onChange={this.handleChange}
									placeholder='enter username'
									value={this.state.username}
									required='required'
								/>
								<InputField
									type='password'
									name='password'
									label='Password'
									required='required'
									onChange={this.handleChange}
									placeholder='enter password'
									value={this.state.password}
								/>
								<button
									type="submit"
									className="btn btn-lg btn-success btn-block btn-signin">
									Login
								</button>
								<p className="message">Not registered? <Link to="/register">Create an account</Link></p>
							</form>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default LoginForm;
