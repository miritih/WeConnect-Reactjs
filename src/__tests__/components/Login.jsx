import React from 'react';
import ReactDOM from 'react-dom';
import Login from '../../Components/Auth/Login';
import { MemoryRouter as Router } from 'react-router-dom';
// import renderer for snapshot testing
import renderer from 'react-test-renderer';

describe('Login', () => {
  it('renders without crashing', () => {
    const pathname = {
      pathname: "/"
    };
    const rendered = renderer.create(
      <Router><Login location={pathname} /></Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
