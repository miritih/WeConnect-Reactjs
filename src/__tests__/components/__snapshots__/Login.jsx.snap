// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders without crashing 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navhome navbar-light bg-transparent fixed-top"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <span
        className="white"
      >
        WeConnect
      </span>
    </a>
    <button
      aria-controls="topnav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#topnav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="topnav"
    >
      <ul
        className="navbar-nav ml-auto white"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        
      </ul>
    </div>
  </nav>
  <div
    className="container wrapper"
  >
    <div
      className="main-center"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <img
            alt="login"
            className="profile-img"
            src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
          />
          <p
            className="profile-name-card"
            id="profile-name"
          />
          <form
            className="form"
            onSubmit={[Function]}
          >
            <label
              htmlFor="username"
            >
              Username
            </label>
            <input
              className="form-control"
              name="username"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label
              htmlFor="password"
            >
              password
            </label>
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              type="text"
              value=""
            />
            <br />
            <button
              className="btn btn-lg btn-success btn-block btn-signin"
              type="submit"
            >
              Login
            </button>
            <p
              className="message"
            >
              Not registered? 
              <a
                href="/register"
                onClick={[Function]}
              >
                Create an account
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
